logging/log.go

```diff
@@ -0,0 +1,82 @@
+package logging
+
+import (
+	"time"
+
+	grpc_logging "github.com/grpc-ecosystem/go-grpc-middleware/logging"
+	grpc_logrus "github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus"
+	"github.com/sirupsen/logrus"
+)
+
+func NewLogger(level string) *logrus.Logger {
+	log := logrus.StandardLogger()
+
+	// Default timeFormat
+	log.SetFormatter(&logrus.JSONFormatter{
+		TimestampFormat: time.RFC3339Nano,
+	})
+
+	// Set the log level on the default log based on command line flag
+	logLevels := map[string]logrus.Level{
+		"debug":   logrus.DebugLevel,
+		"info":    logrus.InfoLevel,
+		"warning": logrus.WarnLevel,
+		"error":   logrus.ErrorLevel,
+		"fatal":   logrus.FatalLevel,
+		"panic":   logrus.PanicLevel,
+	}
+
+	if lvl, ok := logLevels[level]; !ok {
+		log.Errorf("Invalid %q provided for log level", lvl)
+		log.SetLevel(logrus.InfoLevel)
+	} else {
+		log.SetLevel(lvl)
+	}
+
+	return log
+}
+
+var (
+	defaultOptions = &options{
+		levelFunc:    nil,
+		shouldLog:    grpc_logging.DefaultDeciderMethod,
+		codeFunc:     grpc_logging.DefaultErrorToCode,
+		durationFunc: grpc_logrus.DefaultDurationToField,
+		ophIDEnabled: false,
+	}
+)
+
+type options struct {
+	levelFunc    grpc_logrus.CodeToLevel
+	shouldLog    grpc_logging.Decider
+	codeFunc     grpc_logging.ErrorToCode
+	durationFunc grpc_logrus.DurationToField
```

Comment:
I don't see this options can be changed by client, in this case i don't think this is a good idea to add them into options pattern